server/src/main/java/io/crate/analyze/CreateTableStatementAnalyzer.java

```diff
@@ -145,6 +146,14 @@ public AnalyzedCreateTable analyze(CreateTable<Expression> createTable,
             });
         AnalyzedTableElements<Symbol> analyzedTableElementsWithExpressions = TableElementsAnalyzer.analyze(
             tableElementsWithExpressions, relationName, null, false);
+        if (analyzedTableElementsWithExpressions.hasGeneratedColumns()) {
+            List<ColumnIdent> generatedColumns = analyzedTableElementsWithExpressions.columns().stream()
+                .filter(AnalyzedColumnDefinition::isGenerated)
+                .map(AnalyzedColumnDefinition::ident).toList();
+            analyzedTableElementsWithExpressions.columns().stream().filter(AnalyzedColumnDefinition::isGenerated).forEach(
+                s -> GeneratedColumnValidator.validate(s.generatedExpression(), relationName, s.name(), generatedColumns)
+            );
```

Comment:
I think a regular loop would be cheaper, but fine to keep as is

Reply:
Since it's only for DDL, we can keep as is, if you insist, happy to change to a loop.